packages/core/src/internal/utils/dom.ts

```diff
@@ -155,10 +156,20 @@ export function spanWrapper(nodeList: NodeListOf<ChildNode>): void {
     .forEach(node => {
       const spanWrapper = document.createElement('span');
       node.after(spanWrapper);
+      spanWrapper.setAttribute(NAVIGATION_TEXT_WRAPPER, '');
```

Comment:
This seems very specific to the navigation element. Can you think of a way to make it more generalized?

Reply:
Drat, i forgot about this one. I should be adding it in the elements but they are nodes, not elements. This is specific only to the item/start elements when they wrap button  / a text nodes. I'll move it where it is supposed to go and handle adding the attribute there like I meant to.